cli.js

```diff
@@ -57,33 +56,36 @@ function parseCliArguments() {
   return { options, patterns }
 }
 
-function sortPackageJsonFile(file, reporter, isCheck) {
-  const original = fs.readFileSync(file, 'utf8')
+async function sortPackageJsonFile(file, reporter, isCheck) {
+  const original = await fs.readFile(file, 'utf8')
   const sorted = sortPackageJson(original)
   if (sorted === original) {
     return reporter.reportNotChanged(file)
   }
 
   if (!isCheck) {
-    fs.writeFileSync(file, sorted)
+    await fs.writeFile(file, sorted)
   }
 
   reporter.reportChanged(file)
 }
 
-function sortPackageJsonFiles(patterns, { ignore, ...options }) {
-  const files = globSync(patterns, { ignore })
+async function sortPackageJsonFiles(patterns, { ignore, ...options }) {
+  const files = await glob(patterns, { ignore })
 
-  const reporter = new Reporter(files, options)
+  const reporter = new Reporter(options)
   const { isCheck } = options
 
   for (const file of files) {
+    reporter.reportFound(file)
+
     try {
-      sortPackageJsonFile(file, reporter, isCheck)
+      await sortPackageJsonFile(file, reporter, isCheck)
     } catch (error) {
       reporter.reportFailed(file, error)
     }
   }
```

Comment:
Still running serially, hard to make test pass if switch to parallel, I guess we can revist later.